### PR TITLE
docs: in code guidelines, use vi.mocked

### PR DESCRIPTION
**Pull request by feloy**
### What does this PR do?

Create a `CODE-GUIDELINES.md` file, with a first guideline

I would like to visually differentiate the `DON'T` code to the `DO` code, but I have no idea how, I cannot find a simple way to modify code block rendering (reddish background, remove copy button, etc)

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by axel7083 — APPROVED** [review]
LGTM

**Comment by jeffmaury** [comment]
I used the wrong pattern on #10172 but I think it span other parts of the code and we should open an issue to fix it.

Also I remember I had issues if no using a variable but I may check

File: CODE-GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+For example, do not write this, or Typescript won't be able to detect that you passed an object instead of a string to `mockResolvedValue`:
+
+```ts
+const windowMethodMock = vi.fn();
+
+Object.defineProperty(global, 'window', {
+  value: {
+    windowMethod:windowMethodMock,
+  },
+  writable: true,
```
Comment (jeffmaury) [inline CODE-GUIDELINES.md]: Is it required ?

Comment (feloy) [reply CODE-GUIDELINES.md]: for this example, at least, no. I'm removing it

**Review by jeffmaury — APPROVED** [review]
LGTM

**Comment by axel7083** [comment]
Linking https://github.com/podman-desktop/podman-desktop/issues/9821 for reference

**Pull request merged**